src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1021,8 +1021,29 @@ private void InferTypesFrom(CommandAst commandAst, List<PSTypeName> inferredType
             PseudoBindingInfo pseudoBinding = new PseudoParameterBinder()
             .DoPseudoParameterBinding(commandAst, null, null, PseudoParameterBinder.BindingType.ParameterCompletion);
 
-            if (pseudoBinding?.CommandInfo == null)
+            if (pseudoBinding?.CommandInfo is null)
             {
+                var commandName = commandAst.GetCommandName();
+                try
+                {
+                    var foundCommand = CommandDiscovery.LookupCommandInfo(
+                        commandName,
+                        CommandTypes.Application,
+                        SearchResolutionOptions.ResolveLiteralThenPathPatterns,
+                        CommandOrigin.Internal,
+                        _context);
+                    if (foundCommand.CommandType == CommandTypes.Application)
```

Comment:
This check is not needed anymore.

Reply:
Yes but another check was necessary to handle scenario where GetCommandName returns null.

Reply:
Is that needed? I think `CommandDiscovery.LookupCommandInfo` will throw if no result is found.

Reply:
I'm too lazy to make a github link but the method has this right at the start:
```
            if (string.IsNullOrEmpty(commandName))
            {
                return null;
            }
```
but yes, otherwise it throws if it finds nothing.

Reply:
Ah, you're right.